### PR TITLE
Spanish translation

### PR DESCRIPTION
**Pull request by shinmai-rookie**
I've translated the interface into Spanish (if I ever find a somewhat official source I might translate the keywords too). I'm not a player of Magic: The Gathering, so I don't know a lot of the jargon, but I've tried to use the words used in official sources as much as I could.
I use this pull request to propose a (purely hypothetical) bug fix, if you don't mind: I think the strings in the translation file are hashed in such a way that, if the original strings are the same, the hash is also the same, and thus two copies of the same string can't coexist. This works for most sentences, but when used for “P” (abbreviation used for both “played”, in played/won/lost games/duels, and “power”), it creates an equality that doesn't hold for most languages. I suggest that you change one of them to “ P ” or use the invisible whitespace that exists in Unicode to make the strings different.
Kind regards.

### CODE REVIEW

File: release/Magarena/translations/español.txt [hunk 1]
```diff
@@ -0,0 +1,726 @@
...
+3092217981 = Scheme
+2421196991 = Tribal
+2154096770 = Vanguard
+0662641892 = Seleccione sí o no.
+3780066601 = Seleccione una carta de su mano.
+0033616675 = Seleccione %d cartas de su mano.
+1759655335 = Seleccione cara o cruz
+2805059655 = Seleccione un color.
+3462051701 = Seleccione a los atacantes.
+# 0979986174 eg. {f} will be replaced by an icon. | represents a new line.
```
Comment (sblancov) [inline release/Magarena/translations/español.txt]: This line, in spanish, should be:
p. ej. {f} se reemplazará por un icono. | representa una nueva línea.

File: release/Magarena/translations/español.txt [hunk 2]
```diff
@@ -0,0 +1,726 @@
...
+4161141110 = Seleccione una habilidad de maná para pagar %s.
+# 2166000412 eg. %s will be replaced with a mana cost icon.
+2166000412 = No se puede pagar %s.|Pulse {undo} para deshacer.
+3689109508 = Jugar una carta o habilidad.
+# 3864448769 eg. {f} will be replaced by an icon.
+3864448769 = Haga clic en {f} o pulse Espacio para pasar.
+# 2670693515 eg. | represents a new line. Position to fit text in user prompt.
+2670693515 = Haga clic derecho en {f} o pulse Mayús+Espacio|para saltar hasta el fin del turno.
+2959828607 = ¿Desea mover esta carta del tope de su biblioteca al fondo?
+# 0279083863 eg. AI: minimax
```
Comment (sblancov) [inline release/Magarena/translations/español.txt]: In spanish AI is IA, because it comes from Inteligencia Artificial.

File: release/Magarena/translations/español.txt [hunk 3]
```diff
@@ -0,0 +1,726 @@
...
+0279083863 = IA : %s
+1547688846 = Nivel: %d  Vida extra: %d
+# 2291111045 eg. Mini Max, level 6 AI (MMAB)
```
Comment (sblancov) [inline release/Magarena/translations/español.txt]: Same as above, IA instead of AI.

File: release/Magarena/translations/español.txt [hunk 4]
```diff
@@ -0,0 +1,726 @@
...
+2771525700 = El máximo es diez.
+4055741459 = Sacar una
+3615081709 = Aumentar
+4194509842 = Aumentar la cantidad en una unidad, hasta un máximo de diez.
+2069643244 = Reducir
+0391863698 = Reducir la cantidad en una unidad, hasta un mínimo de uno.
+1083993486 = Para eliminar la última copia de una carta debe usar este botón.
+# 0281107096 eg. Memnite <b>is illegal/banned/restricted</b> in Standard.
+0281107096 = %s <b>%s</b> en %s.
+# 3799012839 eg. Illegal deck size for Vintage.
```
Comment (sblancov) [inline release/Magarena/translations/español.txt]: p. ej. Tamaño de mazo ilegal para Vintage.

File: release/Magarena/translations/español.txt [hunk 5]
```diff
@@ -0,0 +1,726 @@
...
+2517211619 = El nombre del mazo no es válido
+1565884469 = Nombre del mazo (debe ser un nombre de archivo válido)
+0303842168 = Guardar el mazo del jugador
+1544943228 = Este mazo es ilegal.\n\n%s
+1884890382 = Deshacer el mazo
+1972294272 = Eliminar todas las cartas del mazo. Requiere confirmación.
+2582893968 = ¿Desea eliminar todas las cartas del mazo?
+3033626837 = Esta acción no se puede deshacer.
+1114478998 = ¿Desea deshacer el mazo?
+# 3448976220 eg. Cards: 290
```
Comment (sblancov) [inline release/Magarena/translations/español.txt]: p. ej. Cartas: 290

Comment (shinmai-rookie) [inline release/Magarena/translations/español.txt]: Thank you for your corrections. I was aware of those sentences not being translated, but, as I understood it, lines beginning with \# are comments, so I simply didn't bother, as they didn't seem to be used for anything beyond explaining the translator what the context of the sentence immediately below was.

Comment (lodici) [inline release/Magarena/translations/español.txt]: @shinmai-rookie many thanks for the spanish translation, I will go ahead and merge it into the main repo. Just to confirm, lines beginning with # do not need to be translated. For more info please see the [Translating Magarena - Comments](https://github.com/magarena/magarena/wiki/Translating-Magarena#comments) section in the wiki.

**Comment by lodici** [comment]
> I use this pull request to propose a (purely hypothetical) bug fix, if you don't mind: I think the strings in the translation file are hashed in such a way that, if the original strings are the same, the hash is also the same, and thus two copies of the same string can't coexist. This works for most sentences, but when used for “P” (abbreviation used for both “played”, in played/won/lost games/duels, and “power”), it creates an equality that doesn't hold for most languages. I suggest that you change one of them to “ P ” or use the invisible whitespace that exists in Unicode to make the strings different.

Good point and thanks for identifying this, I have raised a new issue #1209 address this.

File: release/Magarena/translations/español.txt (hunk 5 above)
Comment (sblancov) [inline release/Magarena/translations/español.txt]: Wop, sorry. Good to know about that rules, I will count with that next time.

**Comment by lodici** [comment]
@shinmai-rookie my apologies, I should have asked whether you were ready to merge you changes first. Having run the latest 1.87 build I noticed that your translation file has significantly more lines than expected. I synchronized your file with the latest build (``Preferences -> español 1.87 -> Edit``) and got the following -
[español.txt](https://github.com/magarena/magarena/files/1164554/espanol.txt)
 
This has significantly less lines than your original file. So I ran a diff between the two and include the results
[diff.zip](https://github.com/magarena/magarena/files/1164558/diff.zip)

Basically it would seem that your original file contains a number of duplicate entries with the same hash code. Magarena does not care and simply selects the first match it finds but that might not be the entry you  meant.

**Comment by shinmai-rookie** [comment]
Yes, you are absolutely right. I am terribly sorry. The bulk of the translation was done for an older version from January, so I had to update it. I added the new strings, but I didn't bother to delete the old ones because I didn't consider that they might have a clashing hash.

Anyway, using some diff magic I managed to get the file to have the same strings and even comments as the original, though not in the same order (about 80 lines are in different places). If that's a problem I can fix it, but it doesn't look like it matters to the program.

The new version:

[español.txt](https://github.com/magarena/magarena/files/1165566/espanol.txt)
